### PR TITLE
Improve handling of Threadmarks migrating between threads.

### PR DESCRIPTION
**Pull request by Xon**
Should help issue #30 by ensuring the threadmark doesn't disappear until the affected threads are made consistent.

### CODE REVIEW

File: upload/library/Sidane/Threadmarks/Install.php [hunk 1]
```diff
@@ -20,6 +20,7 @@ public static function install($existingAddOn, $addOnData)
           post_id INT UNSIGNED NOT NULL,
           label VARCHAR(255) NOT NULL,
           UNIQUE KEY `thread_post_id` (`thread_id`,`post_id`)
```
Comment (sidane) [inline upload/library/Sidane/Threadmarks/Install.php]: Missing comma at the end of this line blows up fresh installations.

**Comment by sidane** [comment]
This is an improvement in that it now shows the threadmarks straight away in the target thread. However, the threadmarks stylesheet isn't loaded on target threads with 0 threadmarks so it looks like this:

Unstyled threadmark

Once the deferred task runs, the stylesheet will load, but in the interim that doesn't look good.

One solution is to always load the CSS on thread pages but that's not ideal as it increases the size of pages without threadmarks needlessly (albeit by a small amount).

Another solution would be to require the CSS in the `threadmarks_render_flag` template instead of through a separate template modification. However, that will mean multiple calls to `xen:require` on pages with multiple threadmarks. XF will only load the CSS once, but it doesn't feel like a very graceful solution.

What do you think @Xon? Any other cleaner solution?

File: upload/library/Sidane/Threadmarks/Install.php (hunk 1 above)
Comment (Xon) [inline upload/library/Sidane/Threadmarks/Install.php]: whoots.

**Comment by Xon** [comment]
I'd actual prefer to include the css in the thread-view itself (or even globally), as it allows better caching on the server all the way to the client as there are less permutations of the CSS file being served.

I would love it if I could just serve a single compiled CSS page and have it used on every page in an XF site. But sadly that doesn't work (I tried).

How xen:require handles duplicates isn't idea (sorting, array_unique, etc), so another template modification to insert it into the thread_view template is my preferred solution.

**Comment by sidane** [comment]
Yeah I like the all up-front CSS approach as well, good for caching. A suggestion for XF 2 perhaps :smile: 

Your latest changes look good :+1:

**Pull request merged**